testng-core-api/src/main/java/org/testng/IObject.java

```diff
@@ -0,0 +1,61 @@
+package org.testng;
+
+import java.util.Objects;
+import java.util.UUID;
+
+/** Represents the associations of a class with one or more instances. */
+public interface IObject {
```

Comment:
Please, try to not expose it in the api

Reply:
@juherr I would either need to add these methods to the `ITestClass` API or have this as a separate interface but these methods would still need to be exposed because we have ended up exposing `ITestClass` interface. 

The other alternative would be to NOT provide alternatives to these methods ( I am of the strong opinion that they dont need to have alternatives) since from the user perspective, all that they will eventually care about is the test class instance associated with a `ITestNGMethod` and that is already available to them. 

So internally we will have `ITestClass` implementations also implement `IObject` interface and we would resort to `is instanceof` checks internally before using these methods.

What do you think of that? I am open to suggestions. The only reason why I introduced these new methods was because I wanted to access the object container instead of the object.

Reply:
Yes. Try an alternative implementation based on instanceof.

I propose to think about open it if someone complains.

Reply:
@juherr - Please check now. I have resorted to using `instanceof` checks